File: shared/dtos/artefact.py
"""Artefact module DTOs."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel

ArtefactType = Literal["markdown", "code", "html", "svg", "jsx", "mermaid"]


class ArtefactSummaryDto(BaseModel):
    id: str
    session_id: str
    handle: str
    title: str
    type: ArtefactType
    language: str | None = None
    size_bytes: int
    version: int
    created_at: datetime
    updated_at: datetime


class ArtefactDetailDto(ArtefactSummaryDto):
    content: str
    max_version: int = 1


class ArtefactListItemDto(BaseModel):
    """Row in the global artefacts list — enriched with session and persona context."""
    id: str
    handle: str
    title: str
    type: ArtefactType
    language: str | None = None
    size_bytes: int
    version: int
    created_at: datetime
    updated_at: datetime
    session_id: str
    session_title: str | None
    persona_id: str
    persona_name: str
    persona_monogram: str
    persona_colour_scheme: str

File: frontend/src/core/types/artefact.ts
export type ArtefactType = 'markdown' | 'code' | 'html' | 'svg' | 'jsx' | 'mermaid'

export interface ArtefactSummary {
  id: string
  session_id: string
  handle: string
  title: string
  type: ArtefactType
  language: string | null
  size_bytes: number
  version: number
  created_at: string
  updated_at: string
}

export interface ArtefactDetail extends ArtefactSummary {
  content: string
  max_version: number
}

export interface ArtefactListItem {
  id: string
  handle: string
  title: string
  type: ArtefactType
  language: string | null
  size_bytes: number
  version: number
  created_at: string
  updated_at: string
  session_id: string
  session_title: string | null
  persona_id: string
  persona_name: string
  persona_monogram: string
  persona_colour_scheme: string
}
